src/sentry/search/events/datasets/spans_metrics.py

```diff
@@ -77,6 +77,31 @@ def resolve_metric(self, value: str) -> int:
             self.builder.metric_ids.add(metric_id)
         return metric_id
 
+    def reflective_result_type(
+        self, index: int = 0
+    ) -> Callable[[list[fields.FunctionArg], dict[str, Any]], str]:
+        """
+        Return the type of the metric, defaults to super implementation
+        """
+
+        def result_type_fn(
+            function_arguments: list[fields.FunctionArg], parameter_values: dict[str, Any]
+        ) -> str:
+            argument = function_arguments[index]
+            value = parameter_values[argument.name]
+            mri = constants.SPAN_METRICS_MAP.get(value)
```

Comment:
Or rather, could we augment get_field_type to handle span metrics, we use that for search field typing as well so the units there need to be correct for the search syntax to handle units correctly. eg. if you search for `some_span_metric:534k` we need to know that its a number to know `k` means thousand

Reply:
hmm makes sense, I originally added a change there but I was updating the `meta_resolver_map` for this field specifically. @markushi I suppose we should port these changes over to that method. I can give that a crack since it's after hours for you right now.

I didn't realize this would only fix the units for the output type and not search

Reply:
@wmak Moved the changes over to the `get_field_type` method in `SpansMetricsQueryBuilder`. Let us know how you feel about it

https://github.com/getsentry/sentry/pull/82209/files#diff-4cf33582e331d118cce575a23d98881984f20b579edcba5122bbcb2a831f0d28R47